lib/checkother.cpp

```diff
@@ -4387,6 +4387,43 @@ void CheckOther::overlappingWriteFunction(const Token *tok, const std::string& f
     reportError(tok, Severity::error, "overlappingWriteFunction", "Overlapping read/write in " + funcname + "() is undefined behavior");
 }
 
+void CheckOther::checkSuspiciousComma()
+{
+    if (!mSettings->severity.isEnabled(Severity::style)) {
+        return;
+    }
+
+    logChecker("CheckOther::suspiciousComma");
+
+    for (const Token* tok = mTokenizer->list.front(); tok; tok = tok->next()) {
+        if (tok->str() == "," && tok->isBinaryOp()) {
+            const Token * parent = tok->astParent();
+            if (parent && Token::Match(parent->previous(), "if|while (")) {
+                if (tok->strAt(-1) == ")") {
+                    const Function * func = tok->linkAt(-1)->previous()->function();
+                    if (func && func->initArgCount > 0 && !tok->astOperand2()->hasKnownValue()) {
```

Comment:
I don't really understand `&& !tok->astOperand2()->hasKnownValue()` .. is there any test that shows why that is needed?

Reply:
I suggest we write both messages. The motivations of the checkers are different.

Maybe this message will be clearer for the developer. He might think that cppcheck writes a alwaysTrueFalse FP because he thinks Cppcheck does not understand the function call properly..